### PR TITLE
Update to use ParseFacebookUtilsV4

### LINKED ISSUE #19: Anywall - problems compiling in Xcode.
Is there some setup I need to do after downloading this resource? I can open the project but it fails to compile. I notice Parse frameworks are not found.

Do I need to do the same setup as I would for my own app to get Parse working?

Otherwise I'm reporting that this release isn't presently working under Xcode (latest up to Jun2nd).

Thanks for any help.

Jules.

### PR DESCRIPTION
**Pull request by html5cat**
Fixes #19 #21 #23

### CODE REVIEW

**Comment by nlutsenko** [comment]
Looks pretty good - will review in depth, but two major things so far:
- auto indent the code using Xcode
- squash all commits into one.

**Comment by html5cat** [comment]
Done and done. Thanks!

**Comment by nlutsenko** [comment]
Please reident using spaces and only spaces.
You can change this setting inside Xcode -> Text Editing -> Indentation.

**Comment by html5cat** [comment]
These are my settings:
screen shot 2015-10-12 at 8 47 07 pm

File: AnyWall-iOS/AnyWall/PAWLoginViewController.m [hunk 1]
```diff
@@ -106,42 +106,31 @@ - (IBAction)loginWithFacebookPressed:(id)sender {
...
+                                           delegate:nil
+                                  cancelButtonTitle:@"Dismiss"
+                                  otherButtonTitles:nil] show];
             }
         } else {
             // Make a call to get user info
-            [FBRequestConnection startForMeWithCompletionHandler:^(FBRequestConnection *connection, NSDictionary<FBGraphUser> *user, NSError *error) {
+
+            FBSDKGraphRequest *request = [[FBSDKGraphRequest alloc]
+                                          initWithGraphPath:@"/me"
```
Comment (nlutsenko) [inline AnyWall-iOS/AnyWall/PAWLoginViewController.m]: Nit: Move to previous line.

File: AnyWall-iOS/AnyWall/PAWLoginViewController.m [hunk 2]
```diff
@@ -106,42 +106,31 @@ - (IBAction)loginWithFacebookPressed:(id)sender {
...
+                [[[UIAlertView alloc] initWithTitle:alertTitle
+                                            message:alertMessage
+                                           delegate:nil
+                                  cancelButtonTitle:@"Dismiss"
+                                  otherButtonTitles:nil] show];
             }
         } else {
             // Make a call to get user info
-            [FBRequestConnection startForMeWithCompletionHandler:^(FBRequestConnection *connection, NSDictionary<FBGraphUser> *user, NSError *error) {
+
```
Comment (nlutsenko) [inline AnyWall-iOS/AnyWall/PAWLoginViewController.m]: Nit: remove whitespace only line.

File: AnyWall-iOS/AnyWall/PAWLoginViewController.m [hunk 3]
```diff
@@ -106,42 +106,31 @@ - (IBAction)loginWithFacebookPressed:(id)sender {
...
-                if (alertMessage) {
-                    [[[UIAlertView alloc] initWithTitle:alertTitle
-                                                message:alertMessage
-                                               delegate:nil
-                                      cancelButtonTitle:@"Dismiss"
-                                      otherButtonTitles:nil] show];
-                }
+                // If the SDK has a message for the user, surface it.
+                alertTitle = @"Something Went Wrong";
+                alertMessage = error.userInfo[NSLocalizedDescriptionKey] ?: @"Unknown error occured";
```
Comment (nlutsenko) [inline AnyWall-iOS/AnyWall/PAWLoginViewController.m]: Nit: Use `error.localizedDescription`.

File: AnyWall-iOS/AnyWall/PAWLoginViewController.m [hunk 4]
```diff
@@ -153,12 +142,13 @@ - (IBAction)loginWithFacebookPressed:(id)sender {
...
                 } else {
                     // Save the name on Parse
-                    [PFUser currentUser][@"name"] = user.name;
+                    [PFUser currentUser][@"name"] = result[@"name"];
                     [[PFUser currentUser] saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
                         completion();
                     }];
                 }
             }];
+
```
Comment (nlutsenko) [inline AnyWall-iOS/AnyWall/PAWLoginViewController.m]: Nit: Remove whitespace only line.

File: AnyWall-iOS/AnyWall/PAWLoginViewController.m [hunk 5]
```diff
@@ -106,42 +106,31 @@ - (IBAction)loginWithFacebookPressed:(id)sender {
...
             }
         } else {
             // Make a call to get user info
-            [FBRequestConnection startForMeWithCompletionHandler:^(FBRequestConnection *connection, NSDictionary<FBGraphUser> *user, NSError *error) {
+
+            FBSDKGraphRequest *request = [[FBSDKGraphRequest alloc]
+                                          initWithGraphPath:@"/me"
+                                          parameters:@{@"fields": @"name"}];
+            [request startWithCompletionHandler:^(FBSDKGraphRequestConnection *connection,
+                                                  id result,
```
Comment (nlutsenko) [inline AnyWall-iOS/AnyWall/PAWLoginViewController.m]: Nit: Do not line-break inside block declarations.

File: AnyWall-iOS/AnyWall/PAWLoginViewController.m [hunk 6]
```diff
@@ -358,7 +344,7 @@ - (void)setActivityViewVisible:(BOOL)visible {
         activityView.label.text = @"Logging in";
         activityView.label.font = [UIFont boldSystemFontOfSize:20.f];
         [activityView.activityIndicator startAnimating];
-
+
```
Comment (nlutsenko) [inline AnyWall-iOS/AnyWall/PAWLoginViewController.m]: Nit: Remove whitespace.

**Comment by nlutsenko** [comment]
Woohoo, looks perfect!
Please make sure it works before merging.

**Pull request merged**